src/main/kotlin/org/jetbrains/research/testspark/tools/kex/generation/KexProcessManager.kt

```diff
@@ -0,0 +1,350 @@
+package org.jetbrains.research.testspark.tools.kex.generation
+
+import com.github.javaparser.StaticJavaParser
+import com.github.javaparser.ast.NodeList
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration
+import com.github.javaparser.ast.body.FieldDeclaration
+import com.github.javaparser.ast.body.MethodDeclaration
+import com.github.javaparser.ast.stmt.BlockStmt
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.process.OSProcessHandler
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessOutput
+import com.intellij.execution.process.ProcessTerminatedListener
+import com.intellij.openapi.diagnostic.Logger
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.util.Key
+import org.jetbrains.research.testspark.bundles.kex.KexDefaultsBundle
+import org.jetbrains.research.testspark.bundles.kex.KexMessagesBundle
+import org.jetbrains.research.testspark.core.data.TestCase
+import org.jetbrains.research.testspark.core.data.TestGenerationData
+import org.jetbrains.research.testspark.core.generation.llm.getImportsCodeFromTestSuiteCode
+import org.jetbrains.research.testspark.core.monitor.ErrorMonitor
+import org.jetbrains.research.testspark.core.progress.CustomProgressIndicator
+import org.jetbrains.research.testspark.data.CodeType
+import org.jetbrains.research.testspark.data.FragmentToTestData
+import org.jetbrains.research.testspark.data.IJReport
+import org.jetbrains.research.testspark.data.ProjectContext
+import org.jetbrains.research.testspark.data.UIContext
+import org.jetbrains.research.testspark.services.KexSettingsService
+import org.jetbrains.research.testspark.settings.kex.KexSettingsState
+import org.jetbrains.research.testspark.tools.ToolUtils
+import org.jetbrains.research.testspark.tools.kex.KexSettingsArguments
+import org.jetbrains.research.testspark.tools.kex.error.KexErrorManager
+import org.jetbrains.research.testspark.tools.llm.generation.StandardRequestManagerFactory
+import org.jetbrains.research.testspark.tools.template.generation.ProcessManager
+import java.io.File
+import java.io.IOException
+import java.net.URL
+import java.nio.charset.Charset
+import java.util.zip.ZipInputStream
+import kotlin.io.path.Path
+import kotlin.io.path.createDirectories
+
+class KexProcessManager(
+    private val project: Project,
+    private val projectPath: String,
+) : ProcessManager {
+
+    private val kexProcessTimeout: Long = 12000000
+    private val kexErrorManager: KexErrorManager = KexErrorManager()
+    private val log = Logger.getInstance(this::class.java)
+
+    private val kexVersion = KexDefaultsBundle.get("kexVersion")
+    private val kexUrl = "https://github.com/edwin1729/kex/releases/download/$kexVersion/kex-$kexVersion.zip"
```

Comment:
It is better to specify download URL as property as well.
Also, can you explain what are current differences between your Kex fork and the main?

Reply:
Maybe just the version, can be added to the settings? There is now no difference

Reply:
I'm not talking about settings, I'm talking about resources and bundles. URL being hardcoded is not the best solution, so it is better to add it to `KexDefaultsBundle` as well

Reply:
so just the `https://github.com/vorpal-research/kex/releases/download/` part in `KexDefaultsBundle`. I was confused since the url depends on the version